runtime/command/run/run.go

```diff
@@ -87,18 +144,13 @@ func (c *Command) startApp(ctx context.Context, hotServer *hot.Server) error {
 				console.Error(err.Error())
 				return nil
 			}
-			app, err := c.Project.Compile(ctx, c.Flag)
-			if err != nil {
-				console.Error(err.Error())
-				return nil
-			}
-			process, err = app.Start(ctx, listener)
+			p, err := c.compileAndStart(ctx, listener)
 			if err != nil {
 				console.Error(err.Error())
 				return nil
 			}
-			// TODO: host should be dynamic
-			console.Info("Ready on http://127.0.0.1" + c.Port)
+			process = p
```

Comment:
^ Fixes a panic where failures would cause the process to be nil, if the watch command exited, process.Close would panic.